lib/crypto/hashsignature.js

```diff
@@ -66,11 +66,29 @@ function verifySignature(hash, dataSignature, publicKeyId) {
   return extractedPubKeyId.equals(pubKeyId);
 }
 
+/**
+ * @param {Buffer} data
+ * @param {Buffer} dataSignature - compact signature
+ * @param {PublicKey} publicKey
+ * @return {boolean}
+ */
+function verifyDataSignatureWithPubKey(data, dataSignature, publicKey) {
+  var hash = doubleSha(data);
+  var signature = Signature.fromCompact(dataSignature);
+
+  return ECDSA.verify(
+    hash,
+    signature,
+    publicKey
+  );
+}
+
 var hashSignature = {
   signData: signData,
   verifyDataSignature: verifyDataSignature,
   signHash: signHash,
-  verifySignature: verifySignature
+  verifySignature: verifySignature,
+  verifyDataSignatureWithPubKey: verifyDataSignatureWithPubKey
```

Comment:
So much verification methods...

Reply:
I would love to remove the previous ones, but I don't want to break backward compatibility. Well, I guess no one really uses `verifyDataSignature` one, so we can replace it with a new one